File: tests/integration/features/flat-lists/flat-lists.ts
// Principle #12: parameters / arguments / generics render as flat siblings,
// not nested <parameters><parameter/></parameters> wrappers. Each sibling
// carries a field="..." attribute for grouping in queries.

function first<T, U>(a: T, b: U, c: number): T {
    return a;
}

first<string, number>("x", 1, 2);

File: tests/integration/features/flat-lists/flat-lists.java
// Principle #12: flat parameters/arguments/generics. No
// <parameters> wrapper — each parameter is a direct sibling with
// field="parameters".

class FlatLists {
    <T, U extends Comparable<U>> T first(T a, U b, int c) {
        return a;
    }

    void caller() {
        first("x", "y", 1);
    }
}
